src/utils/cache_utils.rs

```diff
@@ -0,0 +1,80 @@
+use crate::caching::cache::Cache;
+use crate::database::dsls::object_dsl::ObjectWithRelations;
+use crate::database::enums::ObjectType;
+use aruna_rust_api::api::storage::services::v2::{
+    CollectionRelations, DatasetRelations, ProjectRelations,
+};
+
+pub fn get_object_children(input: &ObjectWithRelations) -> Vec<String> {
+    Vec::from_iter(
+        input
+            .outbound_belongs_to
+            .0
+            .iter()
+            .filter_map(|r| match r.target_type {
+                ObjectType::OBJECT => Some(r.target_pid.to_string()),
+                _ => None,
+            }),
+    )
+}
+pub fn get_dataset_relations(input: &ObjectWithRelations) -> DatasetRelations {
+    let origin = input.object.id.to_string();
+    let children = get_object_children(input);
+    DatasetRelations {
+        origin,
+        object_children: children,
+    }
+}
+
+pub fn get_collection_children(input: &ObjectWithRelations, cache: &Cache) -> CollectionRelations {
+    let origin = input.object.id;
+    let mut dataset_children: Vec<DatasetRelations> = Vec::new();
+    let object_children = get_object_children(input);
+    for d in input
```

Comment:
Can we please not name variables with a single character? It won't break anyone's neck, but it will make it unnecessarily more difficult for later when you have to touch the code again for some reason. I have already seen this in the PR in several places.